dirsrvtests/tests/suites/retrocl/basic_test.py

```diff
@@ -0,0 +1,120 @@
+#
+# License: GPL (version 3 or any later version).
+# See LICENSE for details.
+# --- END COPYRIGHT BLOCK ---
+
+import logging
+import ldap
+import time
+import pytest
+from lib389.topologies import topology_st as topology
+from lib389.plugins import RetroChangelogPlugin
+from lib389._constants import *
+from lib389.utils import *
+from lib389.tasks import *
+
+USER1_DN = 'uid=user1,'+ DEFAULT_SUFFIX
+USER2_DN = 'uid=user2,'+ DEFAULT_SUFFIX
+USER_PW = 'password'
+ATTR_HOMEPHONE = 'homePhone'
+ATTR_CARLICENSE = 'carLicense'
+
+log = logging.getLogger(__name__)
+
+def test_retrocl_exclude_attr_add(topology):
+    """ Test exclude attribute feature of the retrocl plugin for add operation
+
+    :id: todo
+
+    :setup: Standalone instance
+
+    :steps:
+        1. Enable Retro Changelog
+        2. Enable dynamic plugins
+        3. Add an entry
+        4. Ensure entry attrs are in the changelog
+        5. Exclude an attr
+        6. Add another entry
+        7. Ensure entry attrs are not in the changelog
+
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+        4. Success
+        5. Success
+        6. Success
+        7. Success
+    """
+    st = topology.standalone
+
+    # Enable RetroChangelog plugin.
+    rcl = RetroChangelogPlugin(st)
+    rcl.enable()
+
+    # Set the default targetid.
+    rcl.replace('nsslapd-attribute', 'nsuniqueid:targetUniqueId')
+    # Enable dynamic plugins.
+    rcl.replace('nsslapd-dynamic-plugins', 'on')
+
+    st.restart()
+
+    log.info('Add entry ' + USER1_DN)
+    try:
+        st.add_s(Entry((USER1_DN, {'objectclass': "top person organizationalPerson inetOrgPerson shadowAccount".split(),
+                                  'sn': '1',
+                                  'cn': 'user 1',
+                                  'uid': 'user1',
+                                  'givenname': 'user1',
+                                  'homePhone': '0861234567',
+                                  'carLicense': '131D16674',
+                                  'mail': 'user1@whereever.com',
+                                  'userpassword': USER_PW})))
+    except ldap.LDAPError as e:
+        log.fatal('test_retrocl_exclude_attr_add: Failed to add user' + USER1_DN + ': error ' + e.message['desc'])
+        assert False
+
+    log.info('Verify homePhone and carLicense attrs are in the changelog changestring')
+    cllist = st.search_s('cn=changelog', ldap.SCOPE_SUBTREE, '(targetDn=%s)' % USER1_DN)
+    assert len(cllist) > 0
+    if  cllist[0].hasAttr('changes'):
+        clstr = (cllist[0].getValue('changes')).decode()
+        assert ATTR_HOMEPHONE in clstr
+        assert ATTR_CARLICENSE in clstr
+
+    log.info('Excluding attribute ' + ATTR_HOMEPHONE)
+    cmdline=['/usr/sbin/dsconf', 'standalone1', '-D', DN_DM, '-w', PASSWORD, 'plugin', 'retro-changelog', 'add',
+            '--exclude-attrs', ATTR_HOMEPHONE]
+    log.info('Command used : %s' % cmdline)
```

Comment:
You don't need to use dsconf here.  You should use the same code that the CLI/lib389 would execute.

Reply:
That makes sense alright.